oeps/oep-0046-arch-ddd-domains.rst

```diff
@@ -0,0 +1,469 @@
+  - I want to make money.
+  - I want to do academic research and publish some papers. 
+  - I want to effectively, and easily do my job (course build, course run, learner management, data review).
+    - Effectively includes ability to pick and choose learning components.
+
+Use Cases:
+  **Educator.1** Easy Authoring  *(Core)*:
+
+    - Extensible educational blocks
+    - Course Design (Structure - outline, etc)
+    - Educational Blocks (Teaching components and Assessment)
+
+
+  **Educator.2** Administration of Courses *(Core)*:
+
+    - Launch a course run.
+    - Collect revenue share.
+
+  **Educator.3** Actionable Data and Feedback *(Supporting)*
+
+    - What are my students doing
+
+  **Educator.4** Communities of Practice *(Supporting or Generic?)*
+
+  **Educator.5** Supporting Learners (Community and Mentorship) *(Supporting or Generic?)*
+
+Learners
+--------
+
+  - Consume content to achieve learning outcomes.
+  - Determining learning pathway.
+    - Inside and outside of course.
+
+Motivation:
+  - I want to learn, so that I can improve my life (career change, improve my job, raise, change roles, fulfill educational requirements).
+  - I am a lifelong learner.
+  - I want to connect with others in similar fields of interest.
+
+Use Cases:
+  **Learner.1** Discovery: Find the right course/program to help them achieve their goals *(Supporting or Generic?)*
+
+  **Learner.2** Achieve Learning Outcomes *(Core)*:
+
+    - Rigor/quality of instruction
+
+  **Learner.3** Support through their learning pathway – progress, messaging, social *(Supporting or Generic?)*
+
+  **Learner.4** Proof of Learning, credentials/certifications *(Supporting or Generic?)*
+
+Admins
+------
+
+Motivations:
+  Enterprise Admins:
+    - I want to effectively, and easily do my job (course curation, employee management, data review).
+    - I want to know my time and my employee's time is a good investment.
+    - I want to upskill my employees.
+    - I want to validate completion of internal requirements (legal / business policies).
+    - I want to retain my employees (a benefit).
+    - I want to create a pathway for my employees.
+  Educator Admins:
+    - I want to effectively, and easily do my job (learner management, data review).
+
+Use Cases:
+  **Admin.1** Actionable Data and Feedback *(Supporting)*:
+
+    - Am I getting my money's worth
+
+  **Admin.2** Setting up edX for their org *(Supporting)*:
+
+    - Curation
+    - Integration
+
+Specification
+*************
+
+The Open edX system consists of the following bounded contexts.
+
+Terminology and Behaviors
+=========================
+Subdomain:
+  A group of related bounded contexts. A Subdomain is **Core** if it is a key differentiator for
+  Open edX, **Supporting** if it has Open edX specifics, but isn't **Core**, or **Generic** if
+  it has no Open edX business context.
+
+  There should be **no** synchronous communication between separate subdomains.
```

Comment:
This is a strong statement. Do we want to add language around exceptions, or language like the manifesto uses regarding preference when making design decisions?

Reply:
I'm ok with adding exceptions, but I think it's important that we're making a pretty strong statement of preferences here about what these Subdomains actually imply for architectural decisions.

Reply:
That makes sense. We may want to make a similar statement around not sharing code with business logic across subdomains, or something like that, but this might take more thought.